src/common/users.ts

```diff
@@ -343,10 +345,16 @@ export const getUserReadingRank = async (
     });
   };
 
-  const [{ thisWeek, lastWeek, lastReadTime }, tags] = await Promise.all([
+  const [readingStreakResult, tags] = await Promise.all([
     req.getRawOne<ReadingRankQueryResult>(),
     getReadingTags(),
   ]);
+
+  if (!readingStreakResult) {
+    throw new Error('failed to get user reading streak');
+  }
```

Comment:
This is new, basically it would break if for any reason streak would not exist so at least here now we have dedicated error.